accelerator/cpu_accelerator.py

```diff
@@ -139,7 +139,7 @@ def reset_max_memory_cached(self, device_index=None):
         return
 
     def memory_stats(self, device_index=None):
-        return self.get_rss()
+        return None
```

Comment:
Why return `None`? Is there some other way to observe memory statistics of cpu_accelerator?

Reply:
From the usage of memory_stats caller is expecting a dict or None.   The attribute DeepSpeed is using is `num_alloc_retries` which is GPU specific.
https://github.com/microsoft/DeepSpeed/blob/fb9aebbf254d6d4535e42df75cb08b7c1dbd03f2/deepspeed/runtime/zero/stage3.py#L1895

Any suggestions?  Besides `None`, alternative options is return a dict with attribute `num_alloc_retries` set to 0 and emulate other attributes in the future when needed.

Reply:
Thanks for pointing this out. Here are my thoughts. 
1. We should return `dict` of interesting memory statistics. I think it is okay if keys are different across accelerators.
2. We should change DeepSpeed code to first check `num_alloc_retries` in the returned dict before executing any logic.

What do you think?

Reply:
Updated memory_stats to report `allocated_bytes.all.current` and `allocated_bytes.all.peak`, also check whether key `num_alloc_retries` exist first in use point.